tests/utils.go

```diff
@@ -155,9 +155,12 @@ func CreateClickHouseTestEnvironment(testSet string) (ClickHouseTestEnvironment,
 		Image:        fmt.Sprintf("clickhouse/clickhouse-server:%s", GetClickHouseTestVersion()),
 		Name:         fmt.Sprintf("clickhouse-go-%s-%d", strings.ToLower(testSet), time.Now().UnixNano()),
 		ExposedPorts: []string{"9000/tcp", "8123/tcp", "9440/tcp", "8443/tcp"},
-		WaitingFor: wait.ForAll(wait.ForLog("Ready for connections").WithStartupTimeout(time.Second*time.Duration(120)), wait.ForSQL("9000/tcp", "clickhouse", func(port nat.Port) string {
-			return fmt.Sprintf("clickhouse://default:ClickHouse@localhost:%s", port.Port())
-		})).WithStartupTimeout(time.Second * time.Duration(120)),
+		WaitingFor: wait.ForAll(
+			wait.ForLog("Ready for connections").WithStartupTimeout(time.Second*time.Duration(120)),
+			wait.ForSQL("9000/tcp", "clickhouse", func(port nat.Port) string {
+				return fmt.Sprintf("clickhouse://default:ClickHouse@127.0.0.1:%s", port.Port())
```

Comment:
Changed `localhost` to `127.0.0.1` explicitly, because on IPv6 enabled systems this might be pointing to the IPv6 loopback address `::1`:

```
$ cat /etc/hosts
127.0.0.1        localhost
127.0.1.1        housepaper
::1              localhost ip6-localhost ip6-loopback
ff02::1          ip6-allnodes
ff02::2          ip6-allrouters
```

In that scenario, startup of the container might timeout and ClickHouse only shows very strange error messages.

When debugging, it can be seen that the golang `net.Dialer` prefers the IPv6 address above the IPv4 one, though with the incorrect port. Using `127.0.0.1` makes it explicit IPv4.

![image](https://user-images.githubusercontent.com/954721/194041077-ae3252ad-e764-4fa5-b729-2ae06c6ae160.png)

![image](https://user-images.githubusercontent.com/954721/194041110-06e10a00-4e09-4927-9ede-a1fe18f02de4.png)

Reply:
And I think it's fine for now to use IPv4 explicitly.

I'll see with the Testcontainers project if its possible to create an issue / PR to solve this, specifying whether you want the IPv4 or IPv6 mapped port.

Reply:
https://github.com/testcontainers/testcontainers-go/issues/551